persistence-cassandra/javadsl/src/multi-jvm/scala/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraClusteredPersistentEntitySpec.scala

```diff
@@ -43,6 +43,11 @@ class CassandraClusteredPersistentEntitySpec
     }
     enterBarrier("cassandra-started")
 
+    runOn(node2, node3) {
+      awaitPersistenceInit(system)
+    }
+    enterBarrier("cassandra-awaited")
```

Comment:
This 4-line block is a `noop` because of the code above:


```scala
   runOn(node1) {
      // do stuff
      awaitPersistenceInit(system)
    }
    enterBarrier("cassandra-started")
```

Even if nodes `2` and `3` don't wait on `awaitPersistenceInit(system)` they will block on `enterBarrier("cassandra-started")` and that barrier will only open when `node1`'s persistence is ready.

Reply:
Ok, scratch my comment above.

The important part it that `node2` and `node3` are not actively sending a _PING_ event to ensure their persistence layers are warmed up (which is what `awaitPersistenceInit` is doing under the covers).

This is really unexpected but makes sense to me.

 - - - - - - - -

I think the code can be rewritten into:

```diff
+  // We orchestrate the nodes so only one node pings cassandra concurrently. That node (`node1`)
+  // is responsible for creating the schema: see how `cassandra-journal.keyspace-autocreate`
+  // is disabled for node2 and node3.
runOn(node1) {
  ...
-  awaitPersistenceInit(system)
+  warmupPersistence(system)  //rename the method
}
- enterBarrier("cassandra-started")
+ // Setting up cassandra's schema happens during the warmup of node1's persistence via `keyspace-autocreate`
+ enterBarrier("cassandra-schema-setup")

-    runOn(node2, node3) { 
-      awaitPersistenceInit(system)
+      warmupPersistence(system)  //rename the method
-    }
-    enterBarrier("cassandra-awaited")
+    enterBarrier("cassandra-persistence-ready")

```

There's a couple of barrier renames and also a couple of comments to make it more obvious how this whole orchestration works.

Nice find!

Reply:
Yeah, all glory to @chbatey.  I added some comments and changed up the barrier names, but I kept the `awaitPersistenceInit` name as it has ties to APC (for which I added a comment).